Frame pool leaking goroutine through time.Sleep

Seeing failures on travis sometimes due to leaking goroutines in state sleep.

The stack trace shows that the NewSyncFramePool is calling time.Sleep -- which doesn't make sense since there's no calls to sleep in the New function.

```
--- FAIL: TestCloseSingleChannel (0.02s)
	verify.go:121: Found leaked goroutine: Goroutine 191 in state sleep:
		goroutine 191 [sleep]:
		time.Sleep(0x186a0)
			/tmp/workdir/go/src/runtime/time.go:59 +0xf9
		github.com/uber/tchannel-go.NewSyncFramePool.func1(0x0, 0x0)
			/home/travis/gopath/src/github.com/uber/tchannel-go/frame_pool.go:52 +0xab
		sync.(*Pool).Get(0xc820014b00, 0x0, 0x0)
			/tmp/workdir/go/src/sync/pool.go:96 +0x46
		github.com/uber/tchannel-go.syncFramePool.Get(0xc820014b00, 0x8)
			/home/travis/gopath/src/github.com/uber/tchannel-go/frame_pool.go:57 +0x2f
		github.com/uber/tchannel-go.(*syncFramePool).Get(0xc82002e048, 0xc820104600)
			<autogenerated>:205 +0xb4
		github.com/uber/tchannel-go.(*Connection).readFrames(0xc8206ebe00, 0x20)
			/home/travis/gopath/src/github.com/uber/tchannel-go/connection.go:653 +0x74
		created by github.com/uber/tchannel-go.(*Channel).newConnection
			/home/travis/gopath/src/github.com/uber/tchannel-go/connection.go:267 +0x135a
	verify.go:127: Expect at most 2 goroutines, found more:
		[Goroutine 1 in state chan receive:
		goroutine 1 [chan receive]:
		testing.RunTests(0xaea5b0, 0xc8f9a0, 0x6b, 0x6b, 0xcb5e01)
			/tmp/workdir/go/src/testing/testing.go:562 +0xafa
		testing.(*M).Run(0xc82034cf10, 0xc820013d90)
			/tmp/workdir/go/src/testing/testing.go:494 +0xe5
		main.main()
			github.com/uber/tchannel-go/_test/_testmain.go:286 +0x210
		
		 Goroutine 17 in state syscall, locked to thread:
		goroutine 17 [syscall, locked to thread]:
		runtime.goexit()
			/tmp/workdir/go/src/runtime/asm_amd64.s:1696 +0x1
		
		 Goroutine 191 in state sleep:
		goroutine 191 [sleep]:
		time.Sleep(0x186a0)
			/tmp/workdir/go/src/runtime/time.go:59 +0xf9
		github.com/uber/tchannel-go.NewSyncFramePool.func1(0x0, 0x0)
			/home/travis/gopath/src/github.com/uber/tchannel-go/frame_pool.go:52 +0xab
		sync.(*Pool).Get(0xc820014b00, 0x0, 0x0)
			/tmp/workdir/go/src/sync/pool.go:96 +0x46
		github.com/uber/tchannel-go.syncFramePool.Get(0xc820014b00, 0x8)
			/home/travis/gopath/src/github.com/uber/tchannel-go/frame_pool.go:57 +0x2f
		github.com/uber/tchannel-go.(*syncFramePool).Get(0xc82002e048, 0xc820104600)
			<autogenerated>:205 +0xb4
		github.com/uber/tchannel-go.(*Connection).readFrames(0xc8206ebe00, 0x20)
			/home/travis/gopath/src/github.com/uber/tchannel-go/connection.go:653 +0x74
		created by github.com/uber/tchannel-go.(*Channel).newConnection
			/home/travis/gopath/src/github.com/uber/tchannel-go/connection.go:267 +0x135a
		]
```

--- a/testutils/goroutines/verify.go
+++ b/testutils/goroutines/verify.go
@@ -77,16 +77,32 @@ func filterStacks(stacks []Stack, opts *VerifyOpts) []Stack {
 	return filtered
 }
 
+var retryStates = map[string]struct{}{
+	"runnable": struct{}{},
+	"running":  struct{}{},
+	"syscall":  struct{}{},
+}
+
+func shouldRetry(stack Stack) bool {
+	if _, ok := retryStates[stack.State()]; ok {
+		return true
+	}
+
+	if stack.State() == "sleep" {
+		// Go 1.5 has a time.Sleep in the GC path. If we find the exact
+		// time.Sleep call, we should treat this as a runnable goroutine.
+		// see: https://github.com/uber/tchannel-go/issues/151
+		return bytes.Contains(stack.Full(), []byte("time.Sleep(0x186a0)"))
+	}
+
+	return false
+}
+
 // VerifyNoLeaks verifies that there are no goroutines running that are stuck
 // inside of readFrames or writeFrames.
 // Since some goroutines may still be performing work in the background, we retry the
 // checks if any goroutines are fine in a running state a finite number of times.
 func VerifyNoLeaks(t *testing.T, opts *VerifyOpts) {
-	retryStates := map[string]struct{}{
-		"runnable": struct{}{},
-		"running":  struct{}{},
-		"syscall":  struct{}{},
-	}
 	const maxAttempts = 50
 	var leakAttempts int
 	var stacks []Stack
@@ -97,7 +113,7 @@ retry:
 		stacks = GetAll()[1:]
 		stacks = filterStacks(stacks, opts)
 		for _, stack := range stacks {
-			if _, ok := retryStates[stack.State()]; ok {
+			if shouldRetry(stack) {
 				runtime.Gosched()
 				if i > maxAttempts/2 {
 					time.Sleep(time.Millisecond)

